### ISSUE TITLE
ClientV2 not propagating CREATE TABLE exceptions

### ISSUE THREAD

**Issue by alxhill** [ClickHouse/clickhouse-java#1885]
We have a test that tries to create a table from an empty file:

```
CREATE TABLE {tmpTableName:Identifier}  ENGINE=Log() AS SELECT * FROM s3('...empty.csv', 'CSVWithNames')
```

In the previous client, this test correctly throws an exception:

```java
com.clickhouse.client.ClickHouseException: Code: 636. DB::Exception: The table structure cannot be extracted from a CSVWithNames format file. Error:
The table structure cannot be extracted from a CSVWithNames format file: the file is empty.
You can specify the structure manually. (CANNOT_EXTRACT_TABLE_STRUCTURE) (version 24.7.3.42 (official build))

	at com.clickhouse.client.ClickHouseException.of(ClickHouseException.java:151) ~[clickhouse-jdbc-0.6.5.jar:clickhouse-jdbc 0.6.5 (revision: cbdbab0)]
	at com.clickhouse.client.AbstractClient.execute(AbstractClient.java:282) ~[clickhouse-jdbc-0.6.5.jar:clickhouse-jdbc 0.6.5 (revision: cbdbab0)]
	at com.clickhouse.client.ClickHouseClientBuilder$Agent.sendOnce(ClickHouseClientBuilder.java:282) ~[clickhouse-jdbc-0.6.5.jar:clickhouse-jdbc 0.6.5 (revision: cbdbab0)]
	at com.clickhouse.client.ClickHouseClientBuilder$Agent.send(ClickHouseClientBuilder.java:294) ~[clickhouse-jdbc-0.6.5.jar:clickhouse-jdbc 0.6.5 (revision: cbdbab0)]
	at com.clickhouse.client.ClickHouseClientBuilder$Agent.execute(ClickHouseClientBuilder.java:349) ~[clickhouse-jdbc-0.6.5.jar:clickhouse-jdbc 0.6.5 (revision: cbdbab0)]
	at com.clickhouse.client.ClickHouseRequest.execute(ClickHouseRequest.java:2144) ~[clickhouse-jdbc-0.6.5.jar:clickhouse-jdbc 0.6.5 (revision: cbdbab0)]
	at com.clickhouse.client.api.Client.lambda$query$10(Client.java:1402) ~[client-v2-0.6.5.jar:client-v2 0.6.5 (revision: cbdbab0)]
	at com.clickhouse.client.api.Client.runAsyncOperation(Client.java:1624) ~[client-v2-0.6.5.jar:client-v2 0.6.5 (revision: cbdbab0)]
	at com.clickhouse.client.api.Client.query(Client.java:1414) ~[client-v2-0.6.5.jar:client-v2 0.6.5 (revision: cbdbab0)]
```

But in ClientV2, no exception is thrown but the table doesn't get created - instead, it throws when calling `.getTableSchema()`:

```
Caused by: com.clickhouse.client.api.ClientException: Query request failed
	at com.clickhouse.client.api.Client.lambda$query$11(Client.java:1535)
	at com.clickhouse.client.api.Client.runAsyncOperation(Client.java:1864)
	at com.clickhouse.client.api.Client.query(Client.java:1572)
	at io.nominal.csvsplitter.pivot.clickhouse.NominalClickHouseImpl.querySync(NominalClickHouseImpl.java:70)
	at io.nominal.csvsplitter.pivot.clickhouse.NominalClickHouse.querySync(NominalClickHouse.java:45)
	at io.nominal.csvsplitter.pivot.clickhouse.NominalClickHouseImpl.getTableSchema(NominalClickHouseImpl.java:55)
	... 122 more
Caused by: com.clickhouse.client.api.ServerException: Code: 60. DB::Exception: Table db.tmpTableName does not exist. (UNKNOWN_TABLE) (version 24.7.3.42 (official build))
	at com.clickhouse.client.api.internal.HttpAPIClientHelper.readError(HttpAPIClientHelper.java:314)
	at com.clickhouse.client.api.internal.HttpAPIClientHelper.executeRequest(HttpAPIClientHelper.java:357)
	at com.clickhouse.client.api.Client.lambda$query$11(Client.java:1501)
	... 127 more
```

### Code example
```java
        try {
            nominalClickHouse.executeCommandSync(
                    "CREATE TABLE {tmpTableName:Identifier} ON CLUSTER clickhouse ENGINE=Log() AS SELECT * FROM ...",
                    Map.of("tmpTableName", tempTableSettings.tableName()),
                    tempTableSettings.settings());
        } catch (RuntimeException e) {
            log.info(
                    "Exception while creating temporary table, attempting to delete {}",
                    tempTableSettings.tableName(),
                    e);
            // ... etc
        }

```

**Comment by chernser** [comment]
@alxhill thank you for reporting! 
I think, it should be fixed recently - will check.

**Comment by chernser** [comment]
Good day, @alxhill!
 would you please re-test with the latest version 0.7.1? I did not reproduce the issue and client has thrown `ServerException`: 
 
```java

        try {
            client.execute("CREATE TABLE emptyTableName  ENGINE=Log() AS SELECT * FROM url('http://localhost:8000/empty.csv')\n")
                    .get();
            fail("Should have thrown an exception");
        } catch (ServerException e) { 

            // for synchronous request 
            Assert.assertEquals(e.getCode(), 636);
        } catch (ExecutionException e) {
           // if async turned on and request executed in separate thread
            Assert.assertTrue(e.getCause() instanceof ServerException);
        }
```

It may be one of two issues:
- server returns 200 OK because it needs to respond with headers but server doesn't know if operation will be successful. This issue is fixed in 0.7.1 on client 
- exception handling of async operation. For that I need to know how `executeCommandSync` works. 

Thanks!

**Comment by alxhill** [comment]
Yep this is fixed in 0.7.1! The exceptions are missing some info - created #1906 to track 🙂

Thanks!
